src/modules/krita/producer_krita.c

```diff
@@ -0,0 +1,224 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <limits.h>
+
+#include <framework/mlt_frame.h>
+#include <framework/mlt_producer.h>
+#include <framework/mlt_service.h>
+#include <framework/mlt_factory.h>
+#include <framework/mlt_property.h>
+
+
+
+typedef struct
+{
+	int frame_start;
+	int frame_end;
+	int limit_enabled;
+	double speed;
+	mlt_producer producer_internal;
+} private_data;
+
+
+static int restrict_range( int input, int min, int max ) {
+	const int span = max - min;
+	return (MAX(input - min, 0) % (span + 1)) + min;
+}
+
+
+int is_valid_range( const private_data* pdata ) {
+	return pdata->frame_start > -1 && pdata->frame_end > -1 && pdata->frame_end > pdata->frame_start;
+}
+
+int is_limit_enabled( const private_data* pdata) {
+	return pdata->limit_enabled != 0;
+}
+
+static int producer_get_audio( mlt_frame frame, void** buffer, mlt_audio_format* format, int* frequency, int* channels, int* samples )
+{
+	mlt_producer producer = mlt_frame_pop_audio( frame );
+	private_data* pdata = (private_data*)producer->child;
+	struct mlt_audio_s audio;
+	mlt_audio_set_values( &audio, *buffer, *frequency, *format, *samples, *channels );
+
+	int error = mlt_frame_get_audio( frame, &audio.data, &audio.format, &audio.frequency, &audio.channels, &audio.samples );
+
+	// Scale the frequency to account for the speed change.
+	audio.frequency = (double)audio.frequency * fabs(pdata->speed);
+
+	if( pdata->speed < 0.0 )
+	{
+		mlt_audio_reverse( &audio );
+	}
+
+	mlt_audio_get_values( &audio, buffer, frequency, format, samples, channels );
+
+	return error;
+}
+
+static int producer_get_frame( mlt_producer producer, mlt_frame_ptr frame, int index )
+{
+	private_data* pdata = (private_data*)producer->child;
+
+	const int position = mlt_producer_position(pdata->producer_internal);
+	if (is_valid_range(pdata) && is_limit_enabled(pdata)) {
+		mlt_properties_set_position(MLT_PRODUCER_PROPERTIES(pdata->producer_internal), "_position", restrict_range(position, pdata->frame_start, pdata->frame_end));
+	}
+
+	int r = mlt_service_get_frame((mlt_service)pdata->producer_internal, frame, index);
+
+	if ( !mlt_frame_is_test_audio( *frame ) )
+	{
+		mlt_frame_push_audio( *frame, producer );
+		mlt_frame_push_audio( *frame, producer_get_audio );
+	}
+
+	return r;
+}
+
+static int producer_seek( mlt_producer producer, mlt_position position)
+{
+	private_data* pdata = (private_data*)producer->child;
+
+	int r = mlt_producer_seek(pdata->producer_internal, position);
+
+	return r;
+}
+
+static void producer_property_changed( mlt_service owner, mlt_producer self, mlt_event_data event_data)
+{
+	const char *name = mlt_event_data_to_string(event_data);
+	if (!name) return;
+
+	if (strcmp(name, "start_frame") || strcmp( name, "end_frame" )){
```

Comment:
`strcmp()`  returns 0 for equality, which is boolean false in C. So, all string equality tests using it should be preceded with `!` like `if (!strcmp(name, "start_frame") ...`. I am not sure how you get this working for you as-is!

Reply:
Alternately, consider removing this function altogether, remove the properties from the pdata struct and just read the properties directly when you need them. The private data struct is more commonly used to store internal state, not externally set properties.